sdk-python/livescoreapi.py

```diff
@@ -7,17 +7,62 @@
 """
 
 import urllib.request
-
+import json
 
 class livescores:
+    """
+    A class used to get livescores. 
+    ...
+
+    Attributes
+    ----------
+    key : str
+        an alphanumeric string used to access data from the LiveScore API
+    secret : str
+        an alphanumeric string used to access data from the LiveScore API
     
-    def get_all(key,secret):
-        html_file = urllib.request.urlopen("http://livescore-api.com/api-client/scores/live.json?key=" + key + "&secret=" + secret)
-        json_data = html_file.read()
-        return json_data
+    Methods
+    -------
+    _get_html_file(link)
     
-
+    get_all()
+    
+    """
+    link_get_all = "http://livescore-api.com/api-client/scores/live.json?key={}&secret={}"
+   
+    def __init__(self, key, secret):
         
-                
+        try: 
+            key.strip()
+            secret.strip()
+            assert len(key) == 16
+            assert len(secret) == 32
+            self.key = key
+            self.secret = secret 
+            
+        except Exception as e:
+            print("Key or Secret is Empty!")
     
-
+    def _make_api_call(self, link):
+        """returns the source code given a link"""
+    
+        try:
+            http_response_object = urllib.request.urlopen(link)
+            if http_response_object is not None: 
+                json_string = http_response_object.read().decode('utf-8')
+                json_dic = json.loads(json_string)
+                return json_dic
+            
+        except urllib.error.HTTPError as e:
```

Comment:
This method should give an exception to the user so they can handle it the way they want it printing it won't help the user a lot

Reply:
I am not sure what you mean here.  Do you want me to just write the following: 
`except:
            pass`

Reply:
Here you would catch the exception as you do now, and then you will throw a new exception with more meaningfull message like: 'There was a problem connection to the server', you have to check how it will work if the API returns an error code 401 for example if the API key or secret are wrong